DashboardClient/OpcUaTypeReader.cpp

```diff
@@ -12,392 +12,330 @@
+    if (it == notFoundObjectTypeNamespaces.end()) {
+      continue;
+    } else {
+      notFoundObjectTypeNamespaces.erase(it);
+    }
+    findObjectTypeNamespacesAndCreateTypeMap(namespaceURI, bidirectionalTypeMap);
+  }
+
+  for (auto &notFoundObjectTypeNamespace : notFoundObjectTypeNamespaces) {
+    LOG(WARNING) << "Unable to find namespace " << notFoundObjectTypeNamespace;
+  }
+
+  for (auto mapIterator = m_typeMap->begin(); mapIterator != m_typeMap->end(); mapIterator++) {
+    this->updateBaseDataVariableTypes(mapIterator->second);
+  }
+  // printTypeMapYaml();
+  updateTypeMap();
+  updateObjectTypeNames();
+}
+void OpcUaTypeReader::updateObjectTypeNames() {
+  m_expectedObjectTypeNames.clear();
+  for (const auto &el : m_identificationTypeOfTypeDefinition) {
+    auto typeNodeId = el.first;
+    std::shared_ptr<ModelOpcUa::StructureNode> node;
+    try {
+      node = typeDefinitionToStructureNode(typeNodeId);
+      m_expectedObjectTypeNames.push_back(node->SpecifiedBrowseName.Name);
+    } catch (const std::exception &e) {
+      LOG(WARNING) << e.what() << ". Probably because the server does not have the NodeId, specified in the config (" << static_cast<std::string>(typeNodeId)
+                   << ").";
+    }
+  }
+}
+
+void OpcUaTypeReader::initialize(std::vector<std::string> &notFoundObjectTypeNamespaces) {
+  for (auto &m_expectedObjectTypeNamespace : m_expectedObjectTypeNamespaces) {
+    notFoundObjectTypeNamespaces.push_back(m_expectedObjectTypeNamespace);
+  }
+}
+
+void OpcUaTypeReader::printTypeMapYaml() {
+  for (auto mapIterator = m_typeMap->begin(); mapIterator != m_typeMap->end(); mapIterator++) {
+    std::cout << std::endl;
+    ModelOpcUa::StructureNode::printYamlIntern(mapIterator->second, static_cast<std::string>(mapIterator->first), 1, std::cout);
+  }
+}
+void OpcUaTypeReader::updateBaseDataVariableTypes(const std::shared_ptr<ModelOpcUa::StructureNode> &node) {
+  if (std::find(this->BaseDataVariableTypes.begin(), this->BaseDataVariableTypes.end(), node->SpecifiedTypeNodeId) != this->BaseDataVariableTypes.end()) {
+    node->ofBaseDataVariableType = true;
+  }
+  if (!node->SpecifiedChildNodes->empty()) {
+    for (auto childNodesIterator = node->SpecifiedChildNodes->begin(); childNodesIterator != node->SpecifiedChildNodes->end(); childNodesIterator++) {
+      updateBaseDataVariableTypes(*childNodesIterator);
+    }
+  }
+}
+
+void OpcUaTypeReader::updateTypeMap() {
+  for (auto mapIterator = m_typeMap->begin(); mapIterator != m_typeMap->end(); mapIterator++) {
+    for (auto childIterator = mapIterator->second->SpecifiedChildNodes->begin(); childIterator != mapIterator->second->SpecifiedChildNodes->end();
+         childIterator++) {
+      try {
+        auto childTypeNodeId = childIterator->get()->SpecifiedTypeNodeId;
+        if (childTypeNodeId == Dashboard::NodeId_Folder) {
+          for (auto childOfChildIterator = childIterator->get()->SpecifiedChildNodes->begin();
```

Comment:
`for(auto &childOfChild : childIterator->get()->SpecifiedChildNodes)` ? Or is the iterator required?